ansible_base/lib/cache/redis_cache.py

```
@@ -0,0 +1,84 @@
+def optionally_ignore_exceptions(func=None, return_value=None):
+    if func is None:
+        return functools.partial(optionally_ignore_exceptions, return_value=return_value)
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except IGNORED_EXCEPTIONS as e:
+            if getattr(settings, 'DJANGO_REDIS_IGNORE_EXCEPTIONS', True):
+                return return_value
+            raise e.__cause__ or e
```

Ruff (0.15.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)